internal/integrationtest/compile/compile_part_2_test.go

```diff
@@ -190,6 +190,46 @@ func TestCompileWithExportBinariesEnvVar(t *testing.T) {
 	require.FileExists(t, sketchPath.Join("build", fqbn, sketchName+".ino.with_bootloader.hex").String())
 }
 
+func TestCompileWithExportBinariesConfig(t *testing.T) {
+	env, cli := integrationtest.CreateArduinoCLIWithEnvironment(t)
+	defer env.CleanUp()
+
+	// Init the environment explicitly
+	_, _, err := cli.Run("core", "update-index")
+	require.NoError(t, err)
+
+	// Download latest AVR
+	_, _, err = cli.Run("core", "install", "arduino:avr")
+	require.NoError(t, err)
+
+	sketchName := "CompileWithExportBinariesEnvVar"
+	sketchPath := cli.SketchbookDir().Join(sketchName)
+	fqbn := "arduino:avr:uno"
+
+	// Create a test sketch
+	_, _, err = cli.Run("sketch", "new", sketchPath.String())
+	require.NoError(t, err)
+
+	// Create settings with export binaries set to true
+	envVar := cli.GetDefaultEnv()
+	envVar["ARDUINO_SKETCH_ALWAYS_EXPORT_BINARIES"] = "true"
+	_, _, err = cli.RunWithCustomEnv(envVar, "config", "init", "--dest-dir", ".")
+	require.NoError(t, err)
+
+	// Test compilation with export binaries env var set
+	_, _, err = cli.RunWithCustomEnv(envVar, "compile", "-b", fqbn, sketchPath.String())
```

Comment:
In the original test, this execution of the CLI does not set the env var. The comment in the original test is wrong and misleading, it should really be:
```
// Test if arduino-cli config file written in the previous run has the `always_export_binaries` flag set.
```

The goal of this test is to check if the configuration dump by `config init --dest-dir .` contains the value `ARDUINO_SKETCH_ALWAYS_EXPORT_BINARIES` taken from the env var.

Reply:
Yes, it's a standard `compile` run now.